torchmetrics/average.py

```diff
@@ -0,0 +1,24 @@
+from typing import Union
+
+import torch
+
+from .metric import Metric
+
+class AverageMeter(Metric):
+    """Computes the average of a stream of values.
+
+    Forward accepts
+    - ``value`` (scalar): 
+    - ``weight`` (scalar): the weight to give this observation.
+    """
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.add_state("value", torch.zeros(1), dist_reduce_fx="sum")
+        self.add_state("weight", torch.zeros(1), dist_reduce_fx="sum")
+
+    def update(self, value: float, weight: float = 1) -> None:
```

Comment:
since torchmetrics is intended to be used with pytorch, I definitely think that the value should be a `torch.Tensor`. weight can be both tensor and float in my opinion
```suggestion
    def update(self, value: Tensor, weight: Union[float, Tensor] = 1.0) -> None:
```